syntax = "proto3";

package teos.protobuf.protos.v1;

import "google/protobuf/struct.proto";

message RegisterRequest {
  // Requests a user registration with the tower. Contains the user_id in the form of a compressed ecdsa pk
  string user_id = 1;
}

message RegisterResponse {
  // Response to a RegisterRequest, contains the registration information alongside the tower signature of the agreement

  string user_id = 1;
  uint32 available_slots = 2;
  uint32 subscription_expiry = 3;
  string subscription_signature = 4;
}

message GetUserRequest {
  // Request to get information about a specific user. Contains the user id.

  string user_id = 1;
}

message GetUserResponse {
  /* Response with the information the tower has about a specific user. Return is a Struct build from a
  Python dictionary, containing the user's info.
  */

  google.protobuf.Struct user = 1;
}

message GetUsersResponse {
  // Response with information about all the users registered with the tower. Contains a list of user ids.

  repeated string user_ids = 1;
}

message GetSubscriptionInfoRequest {
    // Request to get a specific user's subscription info.

    string signature = 1;
}
